### ISSUE TITLE
internal `TextInput` of `KeyboardExtendedInput` cannot be focused by ref

### ISSUE THREAD

**Issue by proohit** [ArturKalach/react-native-external-keyboard#36]
Hey and thanks for your library. I've encountered an issue while trying to focus the internal react-native TextInput programatically by ref.

So `KeyBoardExtendedInput` does forward a ref, but passes it to `TextInputFocusWrapperNative`. This Component however, does not have a `focus` method, thus I'm getting the below error. This does **not** apply only to keyboard control, but also to touch control. I'm trying to focus the TextInput, after a button has been pressed.

```
const textInputRef = useRef<RNTextInput>(null);
return ( 
   <>
      <Button onPress={() => {textInputRef?.current?.focus?.()}} title="focus"/>
       <KeyboardExtendedInput
          ...
          ref={textInputRef}
      />
   </>
```

I think `KeyboardExtendedInput` should either override `focus` and call it on `TextInput` or the ref should be passed to `TextInput`.
![Screenshot 2024-08-20 at 4 19 16 PM](https://github.com/user-attachments/assets/64b175d9-48fd-4d89-8ee6-f2339966c191)

**Comment by proohit** [comment]
above snippet confirmed to work with this patch:

``` patch
diff --git a/node_modules/react-native-external-keyboard/src/components/KeyboardExtendedInput/KeyboardExtendedInput.tsx b/node_modules/react-native-external-keyboard/src/components/KeyboardExtendedInput/KeyboardExtendedInput.tsx
index 8955709..fd2f338 100644
--- a/node_modules/react-native-external-keyboard/src/components/KeyboardExtendedInput/KeyboardExtendedInput.tsx
+++ b/node_modules/react-native-external-keyboard/src/components/KeyboardExtendedInput/KeyboardExtendedInput.tsx
@@ -53,9 +53,8 @@ export const KeyboardExtendedInput = React.forwardRef<
         focusType={focusMap[focusType]}
         blurType={blurMap[blurType]}
         style={containerStyle}
-        ref={ref}
       >
-        <TextInput blurOnSubmit={false} style={[style, fStyle]} {...props} />
+        <TextInput ref={ref} blurOnSubmit={false} style={[style, fStyle]} {...props} />
       </TextInputFocusWrapperNative>
     );
   }

```

**Comment by ArturKalach** [comment]
Hello @proohit,

______
The issue has been fixed with `react-native-external-keyboard@0.3.5` [version](https://www.npmjs.com/package/react-native-external-keyboard).
______

Thank you so much for creating the issue and sharing the solution; you are totally right. Your solution is exactly what was expected here, but it was mistakenly overlooked and skipped.

If you have any ideas or suggestions, I would really appreciate it if you could share them.


P.S. Right for now, I am preparing a big update with crucial changes (improved focus system, autoFocus, configurable halo effect on iOS, etc.). But right after that, I will add Acknowledgements and appreciations into the Readme.

P.P.S  You know, I'm something of a man of culture myself, and I hope you find more awesome titles soon on your way.

Regards,
Artur Kalach
